scripts/api/deploy.sh

```
@@ -0,0 +1,362 @@
+        *)
+            log "ERROR: Unknown deployment mode: $DEPLOY_MODE"
+            exit 1
+            ;;
+    esac
+}
+
+health_check_service() {
+    case "$DEPLOY_MODE" in
+        binary)
+            if ssh_cmd "pgrep -f basilica-api > /dev/null"; then
+                local port=$(ssh_cmd "grep -E '^(port|bind)' /opt/basilica/config/basilica-api.toml | head -1 | cut -d'=' -f2 | tr -d ' \"'")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)